### PR TITLE
support devcon chip registration + adding new chips

### PR DESCRIPTION
**Pull request by AndrewCLu**
Tap a lanna chip with localhost:3000/tap?chipId=TESTLANNA001

Tap a devcon chip with localhost:3000/tap?encryptedChipId=TESTDEVCON001&cmac=TESTDEVCON001

### CODE REVIEW

File: apps/frontend/public/images/default_header_cover.svg
Comment (stevenelleman) [inline apps/frontend/public/images/default_header_cover.svg:1]: <3

File: apps/frontend/public/images/devcon_register_cover.svg
Comment (stevenelleman) [inline apps/frontend/public/images/devcon_register_cover.svg:1]: <3

**Comment by stevenelleman** [comment]
http://localhost:3000/tap?chipId=TESTDEVCON001 isn't properly redirecting to registration flow. Made sure to install packages and add migrations, but in theory it could be an issue with my setup. Still testing

File: apps/frontend/src/features/register/lanna/CreatingAccount.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  return (
+    <div className="flex flex-col grow">
+      <RegisterHeader
+        title="Setting up your account..."
+        subtitle="Share and connect effortlessly!"
+        description="With a simple tap, you can share your details and make connections instantly—without compromising your privacy. You’re always in control and can update or remove your info whenever you like."
+      />
+    </div>
+  );
+};
```
Comment (stevenelleman) [inline apps/frontend/src/features/register/lanna/CreatingAccount.tsx:14]: Can you pass in some kind of delay on this page? it blipped on my screen for a fraction of a second when I registered a new devcon profile.

File: apps/frontend/src/pages/tap.tsx [hunk 2]
```diff
@@ -129,43 +133,58 @@ const TapPage: React.FC = () => {
...
-
-              toast.success("Successfully bound a new chip to your account");
+          } else if (response.isLocationChip !== true && session) {
+            if (!user?.userData) {
+              toast.error("Please sign in to bind a new chip!");
+              router.push("/login");
+              return;
             }
 
+            const wantsToAddChip = hasRecentAddChipRequest();
```
Comment (stevenelleman) [inline apps/frontend/src/pages/tap.tsx:143]: Nice, clean change.

File: apps/frontend/src/pages/tap.tsx [hunk 3]
```diff
@@ -1,30 +1,35 @@
...
       logClientEvent("tap-chip", {});
-      const { chipId } = router.query;
-
-      if (!chipId || typeof chipId !== "string") {
-        logClientEvent("tap-chip-invalid-chip-id", {});
+      const tapParams = Object.fromEntries(
+        Object.entries(router.query)
+          .filter(([, value]) => typeof value === "string")
+          .map(([key, value]) => [key, value as string])
+      );
```
Comment (stevenelleman) [inline apps/frontend/src/pages/tap.tsx:29]: Does this generalize over both chip types?

**Review by stevenelleman — COMMENTED** [review]
Very clean PR, going to go through all Lanna functionality one more time, just to make sure there are no breaking changes.

**Comment by stevenelleman** [comment]
Hmm I was able to login with a devcon chip without a cmac. Should that be possible?

**Review by stevenelleman — APPROVED** [review]

File: apps/frontend/src/pages/tap.tsx (hunk 3 above)
Comment (AndrewCLu) [reply apps/frontend/src/pages/tap.tsx:29]: yup

**Comment by AndrewCLu** [comment]
> Hmm I was able to login with a devcon chip without a cmac. Should that be possible?

for now yes

**Pull request merged**